<a href="https://colab.research.google.com/github/Yannethtj/PruebasSaberProInteligenciaArtificial/blob/main/01_exploraci%C3%B3n.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import os
os.environ['KAGGLE_CONFIG_DIR'] = '.'
!kaggle
!chmod 600 ./kaggle.json

usage: kaggle [-h] [-v] [-W] {competitions,c,datasets,d,kernels,k,models,m,files,f,config} ...
kaggle: error: the following arguments are required: command


In [3]:
!kaggle competitions download -c udea-ai4eng-20242

 84% 17.0M/20.1M [00:01<00:00, 24.2MB/s]
100% 20.1M/20.1M [00:01<00:00, 18.9MB/s]


In [5]:
!unzip udea-ai4eng-20242.zip

Archive:  udea-ai4eng-20242.zip
  inflating: submission_example.csv  
  inflating: test.csv                
  inflating: train.csv               


In [7]:
import pandas as pd
import numpy as np



In [8]:
train = pd.read_csv("train.csv")
train

ID  PERIODO                  ESTU_PRGM_ACADEMICO  \
0       904256    20212                           ENFERMERIA   
1       645256    20212                              DERECHO   
2       308367    20203                MERCADEO Y PUBLICIDAD   
3       470353    20195           ADMINISTRACION DE EMPRESAS   
4       989032    20212                           PSICOLOGIA   
...        ...      ...                                  ...   
692495   25096    20195                             BIOLOGIA   
692496  754213    20212                           PSICOLOGIA   
692497  504185    20183  ADMINISTRACIÓN EN SALUD OCUPACIONAL   
692498  986620    20195                           PSICOLOGIA   
692499  933374    20195                           PSICOLOGIA   

       ESTU_PRGM_DEPARTAMENTO            ESTU_VALORMATRICULAUNIVERSIDAD  \
0                      BOGOTÁ  Entre 5.5 millones y menos de 7 millones   
1                   ATLANTICO  Entre 2.5 millones y menos de 4 millones   
2                      BOGOTÁ  Entre 2.5 millones y menos de 4 millones   
3                   SANTANDER  Entre 4 millones y menos de 5.5 millones   
4                   ANTIOQUIA  Entre 2.5 millones y menos de 4 millones   
...                       ...                                       ...   
692495             LA GUAJIRA         Entre 500 mil y menos de 1 millón   
692496        NORTE SANTANDER  Entre 2.5 millones y menos de 4 millones   
692497                 BOGOTÁ    Entre 1 millón y menos de 2.5 millones   
692498                 TOLIMA  Entre 2.5 millones y menos de 4 millones   
692499              ANTIOQUIA  Entre 5.5 millones y menos de 7 millones   

       ESTU_HORASSEMANATRABAJA FAMI_ESTRATOVIVIENDA FAMI_TIENEINTERNET  \
0            Menos de 10 horas            Estrato 3                 Si   
1                            0            Estrato 3                 No   
2              Más de 30 horas            Estrato 3                 Si   
3                            0            Estrato 4                 Si   
4          Entre 21 y 30 horas            Estrato 3                 Si   
...                        ...                  ...                ...   
692495     Entre 11 y 20 horas            Estrato 2                 Si   
692496         Más de 30 horas            Estrato 3                 Si   
692497       Menos de 10 horas            Estrato 3                 Si   
692498       Menos de 10 horas            Estrato 1                 No   
692499         Más de 30 horas            Estrato 4                 Si   

                       FAMI_EDUCACIONPADRE  \
0         Técnica o tecnológica incompleta   
1           Técnica o tecnológica completa   
2       Secundaria (Bachillerato) completa   
3                                  No sabe   
4                        Primaria completa   
...                                    ...   
692495  Secundaria (Bachillerato) completa   
692496                 Primaria incompleta   
692497  Secundaria (Bachillerato) completa   
692498                   Primaria completa   
692499  Secundaria (Bachillerato) completa   

                         FAMI_EDUCACIONMADRE ESTU_PAGOMATRICULAPROPIO  \
0                                  Postgrado                       No   
1           Técnica o tecnológica incompleta                       No   
2         Secundaria (Bachillerato) completa                       No   
3         Secundaria (Bachillerato) completa                       No   
4                          Primaria completa                       No   
...                                      ...                      ...   
692495  Secundaria (Bachillerato) incompleta                       Si   
692496  Secundaria (Bachillerato) incompleta                       No   
692497  Secundaria (Bachillerato) incompleta                       Si   
692498                     Primaria completa                       Si   
692499        Técnica o tecnológica completa                       No   

       RENDIMIENTO_GLOB

In [9]:
train.T

0       \
ID                                                                904256   
PERIODO                                                            20212   
ESTU_PRGM_ACADEMICO                                           ENFERMERIA   
ESTU_PRGM_DEPARTAMENTO                                            BOGOTÁ   
ESTU_VALORMATRICULAUNIVERSIDAD  Entre 5.5 millones y menos de 7 millones   
ESTU_HORASSEMANATRABAJA                                Menos de 10 horas   
FAMI_ESTRATOVIVIENDA                                           Estrato 3   
FAMI_TIENEINTERNET                                                    Si   
FAMI_EDUCACIONPADRE                     Técnica o tecnológica incompleta   
FAMI_EDUCACIONMADRE                                            Postgrado   
ESTU_PAGOMATRICULAPROPIO                                              No   
RENDIMIENTO_GLOBAL                                            medio-alto   

                                                                  1       \
ID                                                                645256   
PERIODO                                                            20212   
ESTU_PRGM_ACADEMICO                                              DERECHO   
ESTU_PRGM_DEPARTAMENTO                                         ATLANTICO   
ESTU_VALORMATRICULAUNIVERSIDAD  Entre 2.5 millones y menos de 4 millones   
ESTU_HORASSEMANATRABAJA                                                0   
FAMI_ESTRATOVIVIENDA                                           Estrato 3   
FAMI_TIENEINTERNET                                                    No   
FAMI_EDUCACIONPADRE                       Técnica o tecnológica completa   
FAMI_EDUCACIONMADRE                     Técnica o tecnológica incompleta   
ESTU_PAGOMATRICULAPROPIO                                              No   
RENDIMIENTO_GLOBAL                                                  bajo   

                                                                  2       \
ID                                                                308367   
PERIODO                                                            20203   
ESTU_PRGM_ACADEMICO                                MERCADEO Y PUBLICIDAD   
ESTU_PRGM_DEPARTAMENTO                                            BOGOTÁ   
ESTU_VALORMATRICULAUNIVERSIDAD  Entre 2.5 millones y menos de 4 millones   
ESTU_HORASSEMANATRABAJA                                  Más de 30 horas   
FAMI_ESTRATOVIVIENDA                                           Estrato 3   
FAMI_TIENEINTERNET                                                    Si   
FAMI_EDUCACIONPADRE                   Secundaria (Bachillerato) completa   
FAMI_EDUCACIONMADRE                   Secundaria (Bachillerato) completa   
ESTU_PAGOMATRICULAPROPIO                                              No   
RENDIMIENTO_GLOBAL                                                  bajo   

                                                                  3       \
ID                                                                470353   
PERIODO                                                            20195   
ESTU_PRGM_ACADEMICO                           ADMINISTRACION DE EMPRESAS   
ESTU_PRGM_DEPARTAMENTO                                         SANTANDER   
ESTU_VALORMATRICULAUNIVERSIDAD  Entre 4 millones y menos de 5.5 millones   
ESTU_HORASSEMANATRABAJA                                                0   
FAMI_ESTRATOVIVIENDA                                           Estrato 4   
FAMI_TIENEINTERNET                                                    Si   
FAMI_EDUCACIONPADRE                                              No sabe   
FAMI_EDUCACIONMADRE                   Secundaria (Bachillerato) completa   
ESTU_PAGOMATRICULAPROPIO                                              No   
RENDIMIENTO_GLOBAL                                                  alto   

                                                                  4       \
ID                                

In [13]:
train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 692500 entries, 0 to 692499
Data columns (total 12 columns):
 #   Column                          Non-Null Count   Dtype 
---  ------                          --------------   ----- 
 0   ID                              692500 non-null  int64 
 1   PERIODO                         692500 non-null  int64 
 2   ESTU_PRGM_ACADEMICO             692500 non-null  object
 3   ESTU_PRGM_DEPARTAMENTO          692500 non-null  object
 4   ESTU_VALORMATRICULAUNIVERSIDAD  686213 non-null  object
 5   ESTU_HORASSEMANATRABAJA         661643 non-null  object
 6   FAMI_ESTRATOVIVIENDA            660363 non-null  object
 7   FAMI_TIENEINTERNET              665871 non-null  object
 8   FAMI_EDUCACIONPADRE             669322 non-null  object
 9   FAMI_EDUCACIONMADRE             668836 non-null  object
 10  ESTU_PAGOMATRICULAPROPIO        686002 non-null  object
 11  RENDIMIENTO_GLOBAL              692500 non-null  object
dtypes: int64(2), object(10)
memory

In [14]:
train.columns

Index(['ID', 'PERIODO', 'ESTU_PRGM_ACADEMICO', 'ESTU_PRGM_DEPARTAMENTO',
       'ESTU_VALORMATRICULAUNIVERSIDAD', 'ESTU_HORASSEMANATRABAJA',
       'FAMI_ESTRATOVIVIENDA', 'FAMI_TIENEINTERNET', 'FAMI_EDUCACIONPADRE',
       'FAMI_EDUCACIONMADRE', 'ESTU_PAGOMATRICULAPROPIO',
       'RENDIMIENTO_GLOBAL'],
      dtype='object')

In [15]:

# datos faltantes
k = train.isna().sum()

# Cuantos datos faltantes en cada columna
faltantes = k[k!=0]

print(faltantes)

ESTU_VALORMATRICULAUNIVERSIDAD     6287
ESTU_HORASSEMANATRABAJA           30857
FAMI_ESTRATOVIVIENDA              32137
FAMI_TIENEINTERNET                26629
FAMI_EDUCACIONPADRE               23178
FAMI_EDUCACIONMADRE               23664
ESTU_PAGOMATRICULAPROPIO           6498
dtype: int64


In [16]:
c="ESTU_PRGM_ACADEMICO"
train[c].value_counts()

ESTU_PRGM_ACADEMICO
DERECHO                                                   53244
ADMINISTRACION DE EMPRESAS                                51902
CONTADURIA PUBLICA                                        39664
PSICOLOGIA                                                31932
INGENIERIA INDUSTRIAL                                     28481
                                                          ...  
LICENCIATURA EN ESPAÑOL Y FILOLOGÍA                           1
ADMINISTRACION EN NEGOCIOS INTERNACIONALES                    1
ADMINISTRACION DE COMERCIO EXTERIOR                           1
LICENCIATURA EN EDUCACI¿N F¿SICA RECREACI¿N Y DEPORTES        1
PROFESIONAL EN INVESTIGACIÓN CRIMINAL                         1
Name: count, Length: 948, dtype: int64

Hay un total de 948 filas es decir, hay 948 carreras diferentes.

In [18]:
new1_df = train.groupby(['FAMI_ESTRATOVIVIENDA','RENDIMIENTO_GLOBAL']).size().reset_index(name='NUMERO_DE_ESTUDIANTES')
new_df = train.pivot_table(index='FAMI_ESTRATOVIVIENDA', columns='RENDIMIENTO_GLOBAL', aggfunc='size', fill_value=0)
new_df

RENDIMIENTO_GLOBAL     alto   bajo  medio-alto  medio-bajo
FAMI_ESTRATOVIVIENDA                                      
Estrato 1             14363  42388       23643       31597
Estrato 2             45948  63068       59536       64119
Estrato 3             59121  42756       56841       51967
Estrato 4             28624   9064       16154       11672
Estrato 5             12415   2731        5112        3350
Estrato 6              7648   1287        2314        1356
Sin Estrato             588   1436         579         686

In [20]:

porcentajeFaltante = train.isnull().mean() * 100
print(porcentajeFaltante)


ID                                0.000000
PERIODO                           0.000000
ESTU_PRGM_ACADEMICO               0.000000
ESTU_PRGM_DEPARTAMENTO            0.000000
ESTU_VALORMATRICULAUNIVERSIDAD    0.907870
ESTU_HORASSEMANATRABAJA           4.455884
FAMI_ESTRATOVIVIENDA              4.640722
FAMI_TIENEINTERNET                3.845343
FAMI_EDUCACIONPADRE               3.347004
FAMI_EDUCACIONMADRE               3.417184
ESTU_PAGOMATRICULAPROPIO          0.938339
RENDIMIENTO_GLOBAL                0.000000
dtype: float64


Estamos viendo cual es el porcentaje de datos faltantes en cada columna